### ISSUE TITLE
podman search with "$registry/" should return all images

### ISSUE THREAD

**Issue by thomasmckay** [containers/libpod#1203]
/kind bug

**Description**

To match docker search, podman needs to treat "podman search devel.example.com/" as an indicator to search for all images on the registry. Leaving the trailing slash off, docker returns no results, which I would consider a bug on their part; feel free to implement that too!

**Output of `podman version`:**

```
$ sudo bin/podman version
Version:       0.8.1-dev
Go Version:    go1.9.4
OS/Arch:       linux/amd64

```

**Output of `podman info`:**

```
$ sudo bin/podman info
host:
  MemFree: 827756544
  MemTotal: 8201945088
  SwapFree: 1604804608
  SwapTotal: 1610608640
  arch: amd64
  cpus: 2
  hostname: devel.example.com
  kernel: 3.10.0-862.2.3.el7.x86_64
  os: linux
  uptime: 12h 58m 54.13s (Approximately 0.50 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.centos.org
  - registry.access.redhat.com
store:
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions:
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 0
  RunRoot: /var/run/containers/storage


```

**Comment by mheon** [comment]
@umohnani8 PTAL

**Comment by rhatdan** [comment]
@mtrmac Is this even possible?

**Comment by mtrmac** [comment]
> @mtrmac Is this even possible?

The API should allow it via https://github.com/docker/distribution/blob/master/docs/spec/api.md#listing-repositories , but servers may restrict or not implement this (it does not scale all that well); in particular `docker.io` only supports the V1 registry search mechanism, which requires the caller to submit a search keyword.

**Comment by umohnani8** [comment]
@mtrmac so this is something we want to do for registries that use the V2 API?

**Comment by mtrmac** [comment]
> so this is something we want to do for registries that use the V2 API?

If the goal is compatibility with `docker search` (is it?), I guess so.

**Comment by mheon** [comment]
@mtrmac Yeah, that would be our goal

**Comment by umohnani8** [comment]
Waiting on https://github.com/containers/image/pull/502 to get in and we can vendor in the changes to libpod.

**Comment by umohnani8** [comment]
Being fixed in https://github.com/containers/libpod/pull/1444

**Comment by umohnani8** [comment]
Fixed in #1444
